### PR TITLE
[Add-on PR] Enforce wait for selectIssuers

### PR DESCRIPTION
**Pull request by nambrot**
### Description

Furthers #1327, but opened another PR to make the diff more apparent

###

### CODE REVIEW

File: packages/protocol/contracts/identity/Attestations.sol [hunk 1]
```diff
@@ -779,7 +797,9 @@ contract Attestations is
 
     IRandom random = IRandom(registry.getAddressForOrDie(RANDOM_REGISTRY_ID));
 
-    bytes32 seed = random.random();
+    bytes32 seed = random.getBlockRandomness(
+      unselectedRequest.blockNumber + selectIssuersWaitBlocks
```
Comment (asaj) [inline packages/protocol/contracts/identity/Attestations.sol:799]: Assuming this is what enforces the wait for us?

File: packages/protocol/contracts/identity/Attestations.sol [hunk 2]
```diff
@@ -779,7 +797,9 @@ contract Attestations is
 
     IRandom random = IRandom(registry.getAddressForOrDie(RANDOM_REGISTRY_ID));
 
-    bytes32 seed = random.random();
+    bytes32 seed = random.getBlockRandomness(
```
Comment (asaj) [inline packages/protocol/contracts/identity/Attestations.sol]: `getRandom().getBlockRandomness(`?

File: packages/protocol/contracts/identity/test/MockRandom.sol [hunk 3]
```diff
@@ -1,13 +1,20 @@
 pragma solidity ^0.5.3;
 
 import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "../Random.sol";
 
+contract MockRandom is Random {
+  uint256 public randomnessBlockRetentionWindow = 256;
```
Comment (asaj) [inline packages/protocol/contracts/identity/test/MockRandom.sol]: Where is this used?

File: packages/protocol/contracts/identity/test/MockRandom.sol [hunk 4]
```diff
@@ -1,13 +1,20 @@
...
 
-contract MockRandom {
+  mapping (uint256 => bytes32) private history;
+  uint256 private historyFirst;
```
Comment (asaj) [inline packages/protocol/contracts/identity/test/MockRandom.sol]: ditto

File: packages/protocol/contracts/identity/test/MockRandom.sol [hunk 5]
```diff
@@ -1,13 +1,20 @@
...
+  uint256 private historySize;
```
Comment (asaj) [inline packages/protocol/contracts/identity/test/MockRandom.sol]: ditto

File: packages/protocol/test/identity/attestations.ts [hunk 6]
```diff
@@ -548,6 +592,8 @@ contract('Attestations', (accounts: string[]) => {
 
     beforeEach(async () => {
       await attestations.request(phoneHash, attestationsRequested, mockStableToken.address)
+      const requestBlockNumber = await web3.eth.getBlockNumber()
```
Comment (asaj) [inline packages/protocol/test/identity/attestations.ts]: Can you use `requestAttestations` here?

File: packages/protocol/test/identity/attestations.ts [hunk 7]
```diff
@@ -584,6 +630,8 @@ contract('Attestations', (accounts: string[]) => {
...
```
Comment (asaj) [inline packages/protocol/test/identity/attestations.ts]: ditto

File: packages/protocol/test/identity/attestations.ts [hunk 8]
```diff
@@ -714,6 +762,8 @@ contract('Attestations', (accounts: string[]) => {
     let issuer: string
     beforeEach(async () => {
       await attestations.request(phoneHash, attestationsRequested, mockStableToken.address)
+      const requestBlockNumber = await web3.eth.getBlockNumber()
```
Comment (asaj) [inline packages/protocol/test/identity/attestations.ts]: ditto

**Review by asaj — APPROVED** [review]

File: packages/protocol/contracts/identity/Attestations.sol (hunk 1 above)
Comment (nambrot) [reply packages/protocol/contracts/identity/Attestations.sol:799]: Yup! Implicitly it enforces it assuming that future randomness is not accessible

File: packages/protocol/contracts/identity/test/MockRandom.sol (hunk 3 above)
Comment (nambrot) [reply packages/protocol/contracts/identity/test/MockRandom.sol]: Somehow I thought I ran into an issue and so I just copied the implementation in here, but is fine, so I removed it!

**Comment by nambrot** [comment]
Merging into base branch and then fixing tests from there

**Pull request merged**